supplier_api_tester/supplier_api_tester/utils/availability.py

```diff
@@ -43,6 +43,59 @@ def test_next_30_days(api_url, api_key, product_id, endpoint, adapter_func, vers
 
     return TestResult()
 
+def test_30_days_single_timeslots(api_url, api_key, product_id, endpoint, adapter_func, version=1):
+    '''Checking timeslots'''
+    start = datetime.utcnow().date()
+    end = start + timedelta(days=30)
+
+    raw_response, response = client(f'{api_url}/v{version}/products/{product_id}/{endpoint}', api_key, {
+        'start': start.isoformat(),
+        'end': end.isoformat(),
+    })
+
+    dates_from_response = [result["date"] for result in response]
+    unique_dates_from_response = set(dates_from_response)
+    if len(unique_dates_from_response) == len(dates_from_response):
+        starts_ends = [{"start": item["start"], "end": item["end"]} for item in response]
+        unique =  [i for n, i in enumerate(starts_ends) if i not in starts_ends[n + 1:]]
+        if len(unique) == 1:
+            raise FailedTest(
+            message='Single timeslots cannot be identical every day',
+            response=raw_response,
+        )
+    return TestResult()
+
+
+def test_30_days_timeslots_duplicates(api_url, api_key, product_id, endpoint, adapter_func, version=1):
+    '''Checking timeslots'''
+    start = datetime.utcnow().date()
+    end = start + timedelta(days=30)
+
+    raw_response, response = client(f'{api_url}/v{version}/products/{product_id}/{endpoint}', api_key, {
+        'start': start.isoformat(),
+        'end': end.isoformat(),
+    })
+
+    response_copy = deepcopy(response)
+    for item in response_copy:
+        item.pop("variants")
+
+    unique_timeslots_no_variants = [i for n, i in enumerate(response_copy) if i not in response_copy[n + 1:]]
```

Comment:
This isn't clear to me what's happening here. Also usually `i` is used for the index number with `enumerate`.

Reply:
What I want to do is perform two checks.
First if there are no 'similar' timeslots, meaning - they have the same date, start, end but variants differ - these are not allowed.
I am creating a copy of response containing timeslots, then I am removing variants from the copy.
Then I am making a list with unique records and then I am comparing copy and unique records.
If they are not identical - there must be duplicates.